acceptance/builder_test.go

```diff
@@ -135,14 +135,77 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 			h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
 			h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world")
 			h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.1")
-			h.AssertEq(t, 1, len(md.Processes))
-			h.AssertEq(t, "hello", md.Processes[0].Type)
-			h.AssertEq(t, "echo world", md.Processes[0].Command[0])
-			h.AssertEq(t, 1, len(md.Processes[0].Args))
-			h.AssertEq(t, "arg1", md.Processes[0].Args[0])
-			h.AssertEq(t, false, md.Processes[0].Direct)
-			h.AssertEq(t, "", md.Processes[0].WorkingDirectory)
-			h.AssertEq(t, false, md.Processes[0].Default)
+			h.AssertEq(t, len(md.Processes), 1)
+			h.AssertEq(t, md.Processes[0].Type, "hello")
+			h.AssertEq(t, len(md.Processes[0].Command.Entries), 1)
+			h.AssertEq(t, md.Processes[0].Command.Entries[0], "echo world")
+			h.AssertEq(t, len(md.Processes[0].Args), 1)
+			h.AssertEq(t, md.Processes[0].Args[0], "arg1")
+			h.AssertEq(t, md.Processes[0].Direct, false)
+			h.AssertEq(t, md.Processes[0].WorkingDirectory, "")
+			h.AssertEq(t, md.Processes[0].Default, false)
+		})
+	})
+
+	when("writing metadata.toml", func() {
+		it("writes and reads successfully", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				ctrPath("/layers"),
+				builderImage,
+				h.WithFlags(
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				),
+			)
+			// check builder metadata.toml for success test
+			md := getBuilderMetadata(t, filepath.Join(copyDir, "layers", "config", "metadata.toml"))
+
+			h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
+			h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world")
+			h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.1")
+			h.AssertEq(t, len(md.Processes), 1)
+			h.AssertEq(t, md.Processes[0].Type, "hello")
+			h.AssertEq(t, len(md.Processes[0].Command.Entries), 1)
+			h.AssertEq(t, md.Processes[0].Command.Entries[0], "echo world")
+			h.AssertEq(t, len(md.Processes[0].Args), 1)
+			h.AssertEq(t, md.Processes[0].Args[0], "arg1")
+			h.AssertEq(t, md.Processes[0].Direct, false)
+			h.AssertEq(t, md.Processes[0].WorkingDirectory, "")
+			h.AssertEq(t, md.Processes[0].Default, false)
```

Comment:
Is this redundant to the test above?

Reply:
Could we add a small check like cat'ing out metadata.toml and check that there's no literal newlines?

Reply:
It was the closest test to finding the root problem - I brought in sort of duplicate tests so that the reason/expectations are clearer. I removed a bunch of the assertions in [d3cc6ba](https://github.com/buildpacks/lifecycle/pull/946/commits/d3cc6baa6df958e47430b2f59090fc661fff3f81) on the original test so that the new tests have a clearer definition of what is being checked.

I added `process =` negative test to the contents of the toml file. If that isn't good enough let me know!